### ISSUE TITLE
[4.0] Contacts without Image Description (Alt Text)

### ISSUE THREAD

**Issue by kapetanov** [joomla/joomla-cms#35318]
### Steps to reproduce the issue
1. Open Joomla Backend
2. Create an new Contact (Components > Contacts > New)

### Additional comments
There is no Image Description (Alt Text) field for images

**Comment by brianteeman** [comment]
confirmed @chmst

**Comment by pinak3748** [comment]
The same goes for Banner also (Components > Banners > new) it also doesn't have the image Image Description (Alt Text).

**Comment by brianteeman** [comment]
Note that an alt text is provided for the contact image automatically which is the contact name

**Comment by kapetanov** [comment]
screenreaders read the alt-tag for people with visual impairments or reading difficulties. the contact name can differ from the content of the image.

 example:
 
 a picture of a house, contact name = kindergarten. 
 or a picture of a lot of children, contact name as well as kindergarten.
 
 therefore the alt tag should not be generated from the contact name.

**Comment by chmst** [comment]
Wanted to work on this today. I see that the image is a field with only 255 chars. And that every component has an own treatment of storing images and their params. 
There are several solutions - but each of them needs changes in database and code.

1) Store the image and all it's attributes in the params field (like com_categories). 
Then no image field is needed in #__contact_details.
This is ugly and at least the image field then must be removed from the database, with all the database update scripts in the update routine.

2) Store only the image params in the params field. Probably the simlest solution. No database change but it looks like a quick-and-dirty

3) Change the field image from varchar(255) to text and store the whole image with attributes in the image field. (like com_weblinks). This means a database change.

@richard67 what's  your opinion about that? 



<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/35318">issues.joomla.org/tracker/joomla-cms/35318</a>.</sub>

**Comment by richard67** [comment]
Solution 3 seems to be the cleanest way.

**Comment by charvimehradu** [comment]
hi @chmst,
Can you please explain what you expect by "store the whole image with attributes in the image field". Thankyou in advance.

**Comment by MilanKapetanov** [comment]
hi @chmst,
I think solution 3 is probably a good solution. but best practice would be to save the images in the media manager with the option of alt and description information. a new database table for images would probably be the best way to avoid redundancies when creating new posts. btw. the length of the alt text should not be longer than 140 characters. because screen readers don't read longer alt texts.

**Comment by brianteeman** [comment]
>  btw. the length of the alt text should not be longer than 140 characters. because screen readers don't read longer alt texts.

thats a myth

**Comment by MilanKapetanov** [comment]
@brianteeman 
Alternative text should be succinct, yet descriptive of the content and function of an image. Lengthy alternative text often indicates that extraneous content or content that is not available to sighted users is being presented. Images that present complex content or lengthy text should therefore be included in the image description.

**Comment by brianteeman** [comment]
yes thats absolutely true and I agree. My comment was in reply to your incorrect assertion that scren readers don't read long alt text.

**Comment by MilanKapetanov** [comment]
@brianteeman 
we have checked it years ago with a bunch of screenreader software (nvda, jaws, supernova, cobra ...) and there was in one product a limitation of 140 chars. maybe the new screenreaders software dont have this limitation. people with visual impairments often have older pc systems (because it's expensive), therefore the best practice is probably to set this limitation for alt texts - longer text = description.

**Comment by brianteeman** [comment]
https://yatil.net/blog/there-is-no-character-limit-for-alt-text

**Comment by chmst** [comment]
As much as I know there is not a limit (maybe it was in former years?) but it is recommended to keep alt texts shorter than 140 characters it makes it easier for users of braille lines.

**Comment by brianteeman** [comment]
with Jooa11y we test and recommend at the 100 character limit

**Comment by MilanKapetanov** [comment]
@brianteeman 
there are many screenreader software products on the market. jaws is only one (expensive). and the test was only with jaws. we have checked it (2018 or 2019) with a bunch of products and in one produkt (i think it was nvda) was a limitation. i have now testet it with https://wave.webaim.org/ and it gives me an alert (more then 100 chars). i think 100 chars is now the best.

**Comment by brianteeman** [comment]
>  i think 100 chars is now the best.

I am not disagreeing. Thats why we set the warning in jooa11y at 100 characters

**Comment by brianteeman** [comment]
@MilanKapetanov it's awesome that you clearly have experience with accessibility. It would be great to hear (in a seperate issue) if you have any areas of concern or that can be improved in Joomla. We also have a chat for accessibility on [mattermost](https://joom.la/chat) that you might like to join.

**Comment by MilanKapetanov** [comment]
@brianteeman thank you very much for the invitation, i would be happy to take part in the accessibility talks. we love joomla!
